kong/conf_loader/init.lua

```diff
@@ -141,7 +141,7 @@ local function load_config(thing)
     -- remove trailing comment, if any
     -- and remove escape chars from octothorpes
     if value then
-      value = ngx.re.sub(value, [[\s*(?<!\\)#.*$]], "")
+      value = ngx.re.sub(value, [[\s*(?<!\\)#.*$]], "", "jo")
```

Comment:
> Do not activate the `o` option for regular expressions (and/or `replace` string arguments for [ngx.re.sub](#ngxresub) and [ngx.re.gsub](#ngxregsub)) that are generated *on the fly* and give rise to infinite variations to avoid hitting the specified limit.

According to the [explanation](https://github.com/openresty/lua-nginx-module?tab=readme-ov-file#lua_regex_cache_max_entries) from OpenResty, wouldn't this actually result in poorer performance?

Reply:
This regular expression is not generated on the fly, so it should be Ok.

Reply:
I think that we have talked similar question before, it is OK for `ngx.re.sub` because the replace parameter is static.